File: socketServer.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <arpa/inet.h>

#define PORT 12345
#define BUFFER_SIZE 1024

int main() {
    int server_fd, new_socket;
    struct sockaddr_in address;
    int opt = 1;
    int addrlen = sizeof(address);
    char data[BUFFER_SIZE] = {0};

    // Create socket file descriptor
    server_fd = socket(AF_INET, SOCK_STREAM, 0);

    // Set socket options
    setsockopt(server_fd, SOL_SOCKET, SO_REUSEADDR | SO_REUSEPORT, &opt, sizeof(opt));

    // Bind the socket to the port
    address.sin_family = AF_INET;
    address.sin_addr.s_addr = INADDR_ANY;
    address.sin_port = htons(PORT);
    bind(server_fd, (struct sockaddr *)&address, sizeof(address));
 
    // Start listening for connections
    listen(server_fd, 5);

    printf("Server is listening on port %d...\n", PORT);

    // Accept a connection
    new_socket = accept(server_fd, (struct sockaddr *)&address, (socklen_t *)&addrlen);

    printf("Connection established with client\n");

    // Send a message to the client
    const char *message = "Hello, Client!";
    send(new_socket, message, strlen(message), 0);

    // Receive a message from the client
    int valread = read(new_socket, data, BUFFER_SIZE);
    printf("Received from client: %s\n", data);

    // Close the sockets
    close(new_socket);
    close(server_fd);
}

File: socketServer.py
import socket

# Create a server socket
server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
server_socket.bind(('localhost', 12345))  # Bind to an address and port
server_socket.listen(5)  # Start listening for connections

print(f"Server is listening on port 12345...")

# Accept a connection
client_socket, client_address = server_socket.accept()
print(f"Connection established with {client_address}")

# Send and receive data using client_socket
client_socket.send(b"Hello, Client!")
data = client_socket.recv(1024)
print(f"Received from client: {data}");

# Close the sockets
client_socket.close()
server_socket.close()